### ISSUE TITLE
ra-data-graphql references deprecated version of apollo-client

### ISSUE THREAD

**Issue by lanierc** [marmelab/react-admin#5210]
From the `Readme.md`: Customizing the Apollo Client is explained by showing this:

```import { createNetworkInterface } from 'react-apollo';

buildGraphQLProvider({
    client: {
        networkInterface: createNetworkInterface({
            uri: 'http://api.myproduct.com/graphql',
        }),
    },
});
```

`createNetworkInterface` is deprecated in current versions of `apollo-client`; its last version of support was `react-apollo@1.9.3`, which is also deprecated and no longer available in NPM or Yarn (Source: http://stackoverflow.com/questions/46965564/ddg#46974103)

Either the `ra-data-graphql` needs to be updated to use newer standards in `react-apollo` or the Readme needs to be updated to reflect recent versions of `react-apollo`; as of now this makes following examples to integrate GraphQL in react-admin very difficult.

**Comment by ac2epsilon** [comment]
I see no benefits of referring to any low-level networking, as modern `@apollo/client` does-and-hides it well. Just let's remove this pathway from code and documentation. Better will to say: only way to go is 1. create client 2. create buildQuery 3. do buildGraphQLProvider. This is pretty good for lower level, and should/may be decorated for `simple` layer. Usage looks simple, concise and clean:
```
        const cache = new InMemoryCache();
        const client = new ApolloClient({ cache: cache, uri: 'http://ac2spotters.gq/apollo' });
        buildGraphQLProvider({ client, buildQuery })
          .then(dataProvider => this.setState({ dataProvider }))
          .catch(err => console.log(err));
```
Anyway documentation is subject to change asap, as it's a screamingly disappointing. BTW even if #2934 was closed some years ago, I do not see a mention of the fact (and not see it from code) that `buildQuery` is mandatory. Well, this is logically obvious, but examples ignore it.

**Comment by djhi** [comment]
Fixed by #6614
